sbol_utilities/sbol3_genbank_conversion.py

```diff
@@ -1,48 +1,221 @@
+        )
+        for record in list(SeqIO.parse(gb_file, "genbank").records):
+            # NOTE: Currently we assume only linear or circular topology is possible
+            logging.info(f"Parsing record - {record.id} in genbank file.")
+            if record.annotations["topology"] == "linear":
+                extra_comp_types = [sbol3.SO_LINEAR]
+            else:
+                extra_comp_types = [sbol3.SO_CIRCULAR]
+            comp = sbol3.Component(
+                identity=record.name,
+                types=COMP_TYPES + extra_comp_types,
+                roles=COMP_ROLES,
+                description=record.description,
+            )
+            doc.add(comp)
+            # NOTE: Currently we use a fixed method of encoding (IUPAC)
+            seq = sbol3.Sequence(
+                identity=record.name + "_sequence",
+                elements=str(record.seq.lower()),
+                encoding=SEQUENCE_ENCODING,
+            )
+            doc.add(seq)
+            comp.sequences = [seq]
+            # Sending out warnings for genbank info not storeable in sbol3
+            for keys in record.annotations:
+                logging.warning(
+                    f"Extraneous information not storeable in SBOL3 - {keys}: {record.annotations[keys]}"
+                )
+            if record.features:
+                comp.features = []
+                for gb_feat in record.features:
+                    logging.info(
+                        f"Parsing feature {gb_feat.qualifiers['label'][0]} for record {record.id}"
+                    )
+                    # create "Range/Cut" FeatureLocation by parsing genbank record location
+                    gb_loc = gb_feat.location
+                    # Default orientation is "inline" except if complement is specified via strand
+                    feat_orientation = sbol3.SO_FORWARD
+                    if gb_loc.strand == -1:
+                        feat_orientation = sbol3.SO_REVERSE
+                    # Create a cut or range as featurelocation depending on whether location is specified as
+                    # Cut (eg: "n^n+1", parsed as [n:n] by biopython) or Range (eg: "n..m", parsed as [n:m] by biopython)
+                    if gb_loc.start == gb_loc.end:
+                        locs = sbol3.Cut(
+                            sequence=seq,
+                            at=int(gb_loc.start),
+                            orientation=feat_orientation,
+                        )
+                    else:
+                        locs = sbol3.Range(
+                            sequence=seq,
+                            start=int(gb_loc.start),
+                            end=int(gb_loc.end),
+                            orientation=feat_orientation,
+                        )
+                    # Obtain sequence feature role from gb2so mappings
+                    feat_role = sbol3.SO_NS
```

Comment:
This doesn't feel right to me. Are you sure about this? This will add `'https://identifiers.org/SO:'` to the list of roles? Is that really what you want?

Reply:
No this won't directly get added to feature's roles, as the next if/else would append the exact no after the "SO:" and then the final string would get appended to the feature's roles.
There was one issue though, I must use `sbol3.SO_NS[:-3]`, as we don't want "SO:" to get repeated twice. 
This produces proper feat roles in the test file.